### DISCUSSION TITLE
How to append multiple data arrays into one varibale of xarray dataset?

### DISCUSSION THREAD

**Discussion by hosseinshirali** [pydata/xarray#4921] [Q&A] [answered]
i am new in this field and i need a small help. i just want to know that, what is the best way to append multiple data arrays in a variable of a xarray dataset? 
each data array has a different time and value but has the same x,y coordinates same as dataset. i tried  ```ds[variable_name] = da``` but it works only for the first data array. i want to create a function that gets data arrays and puts them into one variable of the dataset and updating the time dimension of the dataset. thanks for your help

**Comment by keewis** [comment, marked as answer]
you might want to have a look at the relevant [section](https://xarray.pydata.org/en/stable/combining.html) in the documentation. Usually, `xr.concat` with `dim="time"` should be enough, but `xr.combine_by_coords`, `xr.combine_nested` or `xr.merge` could work, too.

**Reply by hosseinshirali** [reply]
yes, ```xr.merge``` solved my problem
thanks
